[Bug]: (*NilPointerError).Error() panics on a nil receiver

### SDK Version

v1.12.0

### Go Version

go1.25.6

### Affected Module / API

core / request pipeline

### What happened?

`internal.NilPointerError`'s `Error()` method dereferences its receiver directly (`return err.s`) with no nil-guard. Calling `.Error()` on a nil `*NilPointerError` panics instead of returning a safe string. Found via a curveball test while adding unit test coverage for `internal/nil_pointer_error.go` (test `TestNilPointerError_Error_NilReceiver`, currently `t.Skip`'d with a repro rather than landed as a passing assertion).

### What did you expect to happen?

`Error()` on a nil receiver should return a safe default string (e.g. `"nil pointer error"`) rather than panicking, consistent with how `conversion.IsNil` guards are used elsewhere in this repo.

### Minimal Reproduction

```go
package main

import "fmt"

func main() {
	var err *NilPointerError // internal.NilPointerError; assume exported for repro
	fmt.Println(err.Error()) // panics
}
```

### Error Output / Stack Trace (if any)

```
panic: runtime error: invalid memory address or nil pointer dereference
```

### Estimated Difficulty (maintainer use)

not triaged yet

### Additional Context

Suggested fix — guard the receiver in `Error()` (e.g. via `conversion.IsNil(err)`) and return a safe default when nil. Related but distinct from #514 (`ServiceNowError.Error()` nil-panic on missing message/detail field).

### Before submitting

- [x] I searched existing issues (including closed ones) for a duplicate of this bug.
- [x] I reproduced this on the latest released version.
- [x] I included a minimal reproduction, not just a description.

--- a/internal/new/nil_pointer_error.go
+++ b/internal/new/nil_pointer_error.go
@@ -14,5 +14,9 @@ func NewNilPointerError(text string) *NilPointerError {
 
 // Error returns the error string.
 func (err *NilPointerError) Error() string {
+	if IsNil(err) {
+		return "nil pointer error"
+	}
+
 	return err.s
 }

--- a/internal/new/nil_pointer_error_test.go
+++ b/internal/new/nil_pointer_error_test.go
@@ -1,0 +1,57 @@
+package internal
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewNilPointerError(t *testing.T) {
+	tests := []struct {
+		name string
+		test func(*testing.T)
+	}{
+		{
+			name: "Successful",
+			test: func(t *testing.T) {
+				err := NewNilPointerError("value is nil")
+
+				assert.Equal(t, "value is nil", err.s)
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, test.test)
+	}
+}
+
+func TestNilPointerError_Error(t *testing.T) {
+	tests := []struct {
+		name string
+		test func(*testing.T)
+	}{
+		{
+			name: "Successful",
+			test: func(t *testing.T) {
+				err := NewNilPointerError("value is nil")
+
+				assert.Equal(t, "value is nil", err.Error())
+			},
+		},
+		{
+			name: "NilReceiver",
+			test: func(t *testing.T) {
+				var err *NilPointerError
+
+				assert.NotPanics(t, func() {
+					assert.Equal(t, "nil pointer error", err.Error())
+				})
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, test.test)
+	}
+}

